### DISCUSSION TITLE
Crashing on macos

### DISCUSSION THREAD

**Discussion by rycossette** [bmaltais/kohya_ss#494] [General]
I'm on an m1 max with 32 gigs of ram. I went through the installation multiple times to make sure I didn't do something wrong. I followed the simple instructions. It installs and runs just fine so it seams. But if I were to try to click on a folder icon to select a checkpoint, or to click on train model, or any button that you'd expect to do something, it crashes. I get this message and I have no clue as to what it means. 

> 2023-03-31 12:36:50.224 python[60884:6320438] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
> *** First throw call stack:
> (
> 	0   CoreFoundation                      0x00000001854603e8 __exceptionPreprocess + 176
> 	1   libobjc.A.dylib                     0x0000000184faaea8 objc_exception_throw + 60
> 	2   CoreFoundation                      0x0000000185489800 _CFBundleGetValueForInfoKey + 0
> 	3   AppKit                              0x000000018868c65c -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 320
> 	4   AppKit                              0x000000018867a94c -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 816
> 	5   AppKit                              0x000000018867a610 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 48
> 	6   libtk8.6.dylib                      0x000000014510a210 TkMacOSXMakeRealWindowExist + 564
> 	7   libtk8.6.dylib                      0x0000000145109e5c TkWmMapWindow + 76
> 	8   libtk8.6.dylib                      0x000000014505eadc Tk_MapWindow + 168
> 	9   libtk8.6.dylib                      0x0000000145067858 MapFrame + 76
> 	10  libtcl8.6.dylib                     0x0000000144f88fd4 TclServiceIdle + 88
> 	11  libtcl8.6.dylib                     0x0000000144f698a0 Tcl_DoOneEvent + 120
> 	12  libtk8.6.dylib                      0x00000001450fbb1c TkpInit + 800
> 	13  libtk8.6.dylib                      0x00000001450601ac Initialize + 2432
> 	14  _tkinter.cpython-310-darwin.so      0x000000014440e284 Tcl_AppInit + 92
> 	15  _tkinter.cpython-310-darwin.so      0x000000014440df20 Tkapp_New + 548
> 	16  _tkinter.cpython-310-darwin.so      0x000000014440dcf8 _tkinter_create_impl + 268
> 	17  _tkinter.cpython-310-darwin.so      0x000000014440d92c _tkinter_create + 280
> 	18  python3.10                          0x0000000102ce274c cfunction_vectorcall_FASTCALL + 88
> 	19  python3.10                          0x0000000102d9abd8 _PyEval_EvalFrameDefault + 40784
> 	20  python3.10                          0x0000000102c84178 _PyFunction_Vectorcall + 548
> 	21  python3.10                          0x0000000102d10b08 slot_tp_init + 332
> 	22  python3.10                          0x0000000102d060f4 type_call + 284
> 	23  python3.10                          0x0000000102db9e68 call_function + 656
> 	24  python3.10                          0x0000000102d9275c _PyEval_EvalFrameDefault + 6868
> 	25  python3.10                          0x0000000102c84178 _PyFunction_Vectorcall + 548
> 	26  python3.10                          0x0000000102dd24f8 _PyObject_VectorcallTstate.4686 + 88
> 	27  python3.10                          0x0000000102dd23bc context_run + 92
> 	28  python3.10                          0x0000000102ce269c cfunction_vectorcall_FASTCALL_KEYWORDS + 84
> 	29  python3.10                          0x0000000102d9c934 _PyEval_EvalFrameDefault + 48300
> 	30  python3.10                          0x0000000102c84178 _PyFunction_Vectorcall + 548
> 	31  python3.10                          0x0000000102d993f8 _PyEval_EvalFrameDefault + 34672
> 	32  python3.10                          0x0000000102c84178 _PyFunction_Vectorcall + 548
> 	33  python3.10                          0x0000000102d993f8 _PyEval_EvalFrameDefault + 34672
> 	34  python3.10                          0x0000000102c84178 _PyFunction_Vectorcall + 548
> 	35  python3.10                          0x0000000102c88944 method_vectorcall + 388
> 	36  python3.10                          0x0000000102e7838c thread_run + 164
> 	37  python3.10                          0x0000000102e18f20 pythread_wrapper + 48
> 	38  libsystem_pthread.dylib             0x000000018530606c _pthread_start + 148
> 	39  libsystem_pthread.dylib             0x0000000185300e2c thread_start + 8
> )
> libc++abi: terminating with uncaught exception of type NSException
> ./gui_macos.sh: line 13: 60884 Abort trap: 6           python kohya_gui.py "$@"

**Comment by Jimmni** [comment]
Same here. Doesn't crash when trying to select a folder (just does nothing), but if I manually input the folder paths then click train, I get a very similar crash and error.


```
Python[37526:801749] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000185f2f154 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x0000000185a4e4d4 objc_exception_throw + 60
	2   CoreFoundation                      0x0000000185f583a0 _CFBundleGetValueForInfoKey + 0
	3   AppKit                              0x00000001890f8a00 -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 320
	4   AppKit                              0x00000001890e6cd4 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 816
	5   AppKit                              0x00000001890e6998 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 48
	6   libtk8.6.dylib                      0x0000000130717550 TkMacOSXMakeRealWindowExist + 524
	7   libtk8.6.dylib                      0x0000000130717234 TkWmMapWindow + 56
	8   libtk8.6.dylib                      0x0000000130689e6c MapFrame + 76
	9   libtcl8.6.dylib                     0x00000001304487ac TclServiceIdle + 84
	10  libtcl8.6.dylib                     0x000000013042c928 Tcl_DoOneEvent + 296
	11  libtk8.6.dylib                      0x000000013070b1cc TkpInit + 712
	12  libtk8.6.dylib                      0x0000000130682de0 Initialize + 2292
	13  _tkinter.cpython-310-darwin.so      0x0000000126d7a334 Tcl_AppInit + 80
	14  _tkinter.cpython-310-darwin.so      0x0000000126d742dc Tkapp_New + 592
	15  _tkinter.cpython-310-darwin.so      0x0000000126d73c8c _tkinter_create + 580
	16  Python                              0x00000001004fab14 cfunction_vectorcall_FASTCALL + 88
	17  Python                              0x0000000100589a90 call_function + 124
	18  Python                              0x0000000100585ff4 _PyEval_EvalFrameDefault + 30060
	19  Python                              0x000000010057db40 _PyEval_Vector + 368
	20  Python                              0x00000001004b69e0 _PyObject_FastCallDictTstate + 96
	21  Python                              0x00000001005180c4 slot_tp_init + 196
	22  Python                              0x0000000100510800 type_call + 284
	23  Python                              0x00000001004b6688 _PyObject_MakeTpCall + 136
	24  Python                              0x0000000100589b24 call_function + 272
	25  Python                              0x0000000100585ff4 _PyEval_EvalFrameDefault + 30060
	26  Python                              0x000000010057db40 _PyEval_Vector + 368
	27  Python                              0x00000001004b69e0 _PyObject_FastCallDictTstate + 96
	28  Python                              0x00000001005180c4 slot_tp_init + 196
	29  Python                              0x0000000100510800 type_call + 284
	30  Python                              0x00000001004b6688 _PyObject_MakeTpCall + 136
	31  Python                              0x0000000100589b24 call_function + 272
	32  Python                              0x000000010058601c _PyEval_EvalFrameDefault + 30100
	33  Python                              0x000000010057db40 _PyEval_Vector + 368
	34  Python                              0x00000001004b6a50 _PyObject_FastCallDictTstate + 208
	35  Python                              0x00000001005180c4 slot_tp_init + 196
	36  Python                              0x0000000100510800 type_call + 284
	37  Python                              0x00000001004b6688 _PyObject_MakeTpCall + 136
	38  Python                              0x0000000100589b24 call_function + 272
	39  Python                              0x0000000100586098 _PyEval_EvalFrameDefault + 30224
	40  Python                              0x000000010057db40 _PyEval_Vector + 368
	41  Python                              0x0000000100589a90 call_function + 124
	42  Python                              0x0000000100586098 _PyEval_EvalFrameDefault + 30224
	43  Python                              0x000000010057db40 _PyEval_Vector + 368
	44  Python                              0x0000000100589a90 call_function + 124
	45  Python                              0x0000000100586098 _PyEval_EvalFrameDefault + 30224
	46  Python                              0x000000010057db40 _PyEval_Vector + 368
	47  Python                              0x0000000100589a90 call_function + 124
	48  Python                              0x0000000100586098 _PyEval_EvalFrameDefault + 30224
	49  Python                              0x000000010057db40 _PyEval_Vector + 368
	50  Python                              0x000000010059c1f8 _PyObject_VectorcallTstate.4691 + 88
	51  Python                              0x000000010059c0b4 context_run + 92
	52  Python                              0x00000001004faa64 cfunction_vectorcall_FASTCALL_KEYWORDS + 84
	53  Python                              0x00000001005861a8 _PyEval_EvalFrameDefault + 30496
	54  Python                              0x000000010057db40 _PyEval_Vector + 368
	55  Python                              0x0000000100589a90 call_function + 124
	56  Python                              0x0000000100585f80 _PyEval_EvalFrameDefault + 29944
	57  Python                              0x000000010057db40 _PyEval_Vector + 368
	58  Python                              0x0000000100589a90 call_function + 124
	59  Python                              0x0000000100585f80 _PyEval_EvalFrameDefault + 29944
	60  Python                              0x000000010057db40 _PyEval_Vector + 368
	61  Python                              0x00000001004b968c method_vectorcall + 404
	62  Python                              0x000000010062310c thread_run + 120
	63  Python                              0x00000001005d61e8 pythread_wrapper + 48
	64  libsystem_pthread.dylib             0x0000000185dd7fa8 _pthread_start + 148
	65  libsystem_pthread.dylib             0x0000000185dd2da0 thread_start + 8
)
libc++abi: terminating due to uncaught exception of type NSException
./gui.sh: line 71: 37526 Abort trap: 6           python "$SCRIPT_DIR/kohya_gui.py" "$@"
```

**Reply by Jimmni** [reply]
I reinstalled repeatedly until the error went away. Seems to have been something not installing properly.

**Reply by bobpuffer1** [reply]
I'm going to give reinstalling some more tries. I've been getting the same error on MBP M2 Max running latest Ventura for the past week. As for the folder not doing anything, I think that's a known issue with Macs.

**Reply by bobpuffer1** [reply]
I was just over in issues and the maintainer states that MacOS isn't working right now and he is not in a position to try to fix it so he's leaving it to the community to try to come up with some solutions. There's one person there that volunteered to try to come up with a macOS solution

**Reply by Jimmni** [reply]
Yeah I got past this issue and then ran into another issue of it running all the taining steps then failing out afterwards. Hopefully someone figures it all out!

**Reply by bobpuffer1** [reply]
What did you do to get past that error?

**Reply by Jimmni** [reply]
I deleted everything, reinstalled a bunch of times, and did a bunch of stuff following various tutorials but I can't remember exact what I'm afraid. There was a tutorial by a Japanese chap that walked through changing "cuda" to "mps" or something similar in one of the files but I don't remember if that actually helped or not. Sorry, I've dug through my history looking for URLs but I must have done it in a private window or something. Some of the things I did even managed to break my Stable-Infusion installation and I had to revert something. At some point after giving up I apparently had to run "pip3 uninstall tensorflow-macos tensorflow-metal tensorboard". Again, no idea if having it installed had actually helped or not.

**Reply by organicmind-pixel** [reply]
i think this link u looking for .. https://planaria-page.translate.goog/blog/?p=671&_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp
